modules/storage/src/service/fs.rs

```diff
@@ -67,14 +73,25 @@ impl FileSystemBackend {
                 )
             })?;
 
-        Ok(Self { content })
+        // the read compressions, have the write compression first, then all remaining
+
+        let mut read_compressions = vec![compression];
+        read_compressions.extend(Compression::iter().filter(|c| c != &compression));
```

Comment:
It took me a sec to understand why you're going to this trouble. Maybe note that the likely read format is the same as the write, so minimize the chance of failed reads?

Reply:
I updated the comment.